rules/order/src/Rector/Class_/OrderPropertyByComplexityRector.php

```diff
@@ -153,4 +158,13 @@ private function resolvePropertyByVisibilityByPosition(Class_ $class): array
 
         return $propertyByVisibilityByPosition;
     }
+
+    private function getSortedPropertiesByRankAndPosition(array $propertyNameToRank): array
+    {
+        $rank = array_column($propertyNameToRank, 'rank');
+        $position = array_column($propertyNameToRank, 'position');
+        array_multisort($rank, SORT_ASC, $position, SORT_ASC, $propertyNameToRank);
+
+        return array_keys($propertyNameToRank);
+    }
```

Comment:
This is very hard read to me. What should it do?

Wouldn't battleship do it?

See the last code snippet here: https://blog.martinhujer.cz/clever-way-to-sort-php-arrays-by-multiple-values/